### ISSUE TITLE
Move Encryptor outside of KeyringController

### ISSUE THREAD

**Issue by danfinlay** [MetaMask/metamask-extension#4088]
Once #4034 is completed, the next function that could be moved outside of the KeyringController is the encryption functions.

Currently the KeyringController stores accounts and their schema, but these aren't actually the only portions of MetaMask state that might need to be encrypted.

Instead, we should move our encryption related functions into their own module at the `MetaMaskController` level, so that it can combine encryption-needing state as needed.

Two functions involving the password currently pass through the KeyringController:
- [ ] setLocked
- [ ] submitPassword

Also, a couple methods that assume keyringController manages the password encryption exist:
- [ ] createNewVaultAndKeychain
- [ ] createNewVaultAndRestore

Since we are using [obs-store](https://www.npmjs.com/package/obs-store) for most of our state objects now, it would probably make sense to take advantage of this architecture for the new encryption strategy.

## One implementation idea: SecureComposableObservableStore

We could maybe achieve this goal with a subclass of [ComposableObservableStore](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/ComposableObservableStore.js), [as the `metamaskController.store` instead](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/metamask-controller.js#L70).

### The `SecureComposableObservableStore` class 

 - Has a method named `subscribeToEncrypted`, which emits an encrypted blob instead of the clear text contents.
- Has a method named `submitPassword`, and does not emit any events until it has been unlocked.
- Has a `setLocked` method that deallocates any secret material and stops emitting updates.
- Emits events for `locked` and `unlocked`.
- `MetaMaskController` uses `locked` and `unlocked` events to setup the controllers managed by this store.
- `MetaMaskController` must also call [`updateStructure(opts)`](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/ComposableObservableStore.js#L24) on its `.memStore` on both of these events, to add/remove the added/removed controllers from the memStore.

**Comment by lazaridiscom** [comment]
You're testing my patience here?

**Comment by danfinlay** [comment]
No. Just recording likely long term architecture changes. The vault may not need to be in charge of encryption. Just ignore if it's distracting.

**Comment by lazaridiscom** [comment]
Whatever. I deleted my trial-project, and removed myself (again) from the Metamask Organization.

As a "Senior Level Code-Clown" I do not need much access rights.

Received my Ledger today, going on with #4060 (as described there).

Good luck with keeping the project on track.
